src/optimizer/rule/simplification.rs

```diff
@@ -109,6 +117,91 @@ impl Rule for SimplifyFilter {
     }
 }
 
+pub struct LikeRewrite;
+
+impl Rule for LikeRewrite {
+    fn pattern(&self) -> &Pattern {
+        &LIKE_REWRITE_RULE
+    }
+
+    fn apply(&self, node_id: HepNodeId, graph: &mut HepGraph) -> Result<(), OptimizerError> {
+        if let Operator::Filter(mut filter_op) = graph.operator(node_id).clone() {
+            // if is like expression
+            if let ScalarExpression::Binary {
+                op: BinaryOperator::Like,
+                left_expr,
+                right_expr,
+                ty,
+            } = &mut filter_op.predicate
+            {
+                // if left is column and right is constant
+                if let ScalarExpression::ColumnRef(_) = left_expr.as_ref() {
```

Comment:
It seems that there is no problem even if left expr is not Cols? e.g. 'ABCC' like 'ABC%' -> true

Reply:
i will fix it